### ISSUE TITLE
Buggy when hitting back button during page transition

### ISSUE THREAD

**Issue by Yvee1** [swup/swup#187]
When one hits the back arrow during a page transition, the result is buggy. On the swup documentation site for example, I can get an empty page with trailing classes on the `html` element.

![screenshot of emtpy page](https://user-images.githubusercontent.com/30909373/63176860-01fbc900-c047-11e9-8648-f2289c42ff82.png)

On the decathlon usage example page, when I hit back during the 'out' transition I do get a non-empty page with no trailing classes, but it is the incorrect page.

![screenshot of page](https://user-images.githubusercontent.com/30909373/63177029-5acb6180-c047-11e9-9735-45a65cf573b1.png)

You can see the url indicates the homepage, but another page is showing.

The problem is the biggest during an out-transition, but during an in-transition on my own page for example there are some trailing classes (`is-changing to-mehtml is-rendering` to be exact) though they don't cause any problems I think.

I do enjoy the library though, really easy to use!

**Comment by gmrchk** [comment]
Hi @Yvee1, thank you for reporting, this has been on my mind for a while. 

Question is, what's the correct approach here. Block back/forward when the transition is in the process? Stop animation and replace the content right away? Some other approach?

I think I would go for a second option.

**Comment by Yvee1** [comment]
I think blocking the buttons is indeed not a good idea. 

I would suggest that when the back button is clicked during an out-transition, all swup classes are removed from the html element hence stopping (or reversing I guess?) the animation, and the url is changed back to what it was before. If the back button is clicked during an in-transition it works fine how it is currently, but maybe all the swup classes on the html element can be removed.

I guess the same would hold for the forward button.

**Comment by idflood** [comment]
Here is a partial fix (when clicking quickly on multiple links):

```javascript
swup.on('clickLink', () => {
  document.documentElement.classList.remove('is-leaving');
  document.documentElement.classList.remove('is-animating');
});
```

**Comment by hirasso** [comment]
This has been addressed in [2.0.18](https://github.com/swup/swup/releases/tag/2.0.18)
